sequencer/src/state.rs

```diff
@@ -543,6 +534,35 @@ impl ValidatedState {
         } else {
             vec![]
         };
+
+        // pushing a block into merkle tree shouldn't fail
+        validated_state
+            .block_merkle_tree
+            .push(parent_leaf.get_block_header().commit())
+            .unwrap();
```

Comment:
This needs to happen after the `remember_blocks_merkle_tree`; that `remember` is the thing that ensures this won't fail.

In general the flow here can be divided into three separate parts:
* Fetch any inputs we need from external sources (currently, just the deposits from L1)
* Fetch any missing data from our own state (missing accounts, block merkle tree frontier if needed)
* Apply state changes (push new block, update fee state based on deposits, charge builder fee). This step cannot fail assuming the previous steps succeeded.

May be a good idea to divide these into three separate functions; then we can use the type system to enforce that e.g. the third step is synchronous (not `async`) and infallible (doesn't return `Result`)